test/ChunkContainer_test.cpp

```diff
@@ -240,6 +240,36 @@ TEST(ChunkContainer, resize)
     }
 }
 
+TEST(ChunkContainer, clear_method)
+{
+    ChunkContainer<int, 8> chk;
+    
+    // populate the ct
+    for (int i = 0; i < 20; ++i)
+    {   
+       chk.push_back(i);
+    }
+    // verify init state
+    EXPECT_THAT(chk.size(), Eq(20u));
+    
+    // clear the ct
+    chk.clear();
+    
+    // check size is reseted 
+    EXPECT_THAT(chk.size(), Eq(0u));
+    
+    // check if we can still add elements after clear
+    for (int i = 0; i < 10; ++i) 
+    {
+       chk.push_back(i);
+    }
+    EXPECT_THAT(chk.size(), Eq(10u));
+    for (int i = 0; i < 10; ++i) 
+    {
+        EXPECT_THAT(chk[size_t(i)], Eq(i)) << "at " << i;
+    }
+}
+
```

Comment:
Thank you for this contribution - a test for this method was indeed needed.

Could you make the test more concise, by doing the following?

1. Use `resize(20, 42)` or something like this to make it size 20 and fill with values `42`.
2. Then clear
3. Then use another `resize` call after and verify size and content. 

For checking content, you can then also use `EXPECT_THAT(chk, Each(Eq(42)))` to test that all elements in the container are equal the value 42. 

I would also verify that after `clear`, the `capacity` does not change (same as before clear).

Reply:
ah yes i forgot about Each matcher

Reply:
done

Reply:
i think i cant use each since its use begin() and end()

Reply:
Yes, true. But for this test, which just tests `clear` method, you don't even need to check the values, just size and capacity (so just remove the checks for the values). 

But please check `capacity` just before `clear` - `resize()` affects `capacity` and that's expected. But `clear` should not affect `capacity`, only `size`.